staging/src/k8s.io/client-go/transport/transport.go

```diff
@@ -76,11 +76,37 @@ func TLSConfigFor(c *Config) (*tls.Config, error) {
 	}
 
 	if c.HasCertAuth() {
-		cert, err := tls.X509KeyPair(c.TLS.CertData, c.TLS.KeyData)
-		if err != nil {
+		// If key/cert were provided, verify them before setting up
+		// tlsConfig.GetClientCertificate.
+		if _, err := tls.X509KeyPair(c.TLS.CertData, c.TLS.KeyData); err != nil {
```

Comment:
save the result of this, and return that from GetClientCertificate as the fallback case rather than reparsing the certificate on every call to GetClientCertificate

Reply:
Done